### ISSUE TITLE
Using PIMixin disables ModelicaMixin's initial_state

### ISSUE THREAD

**Issue by SGeeversAtVortech** [Deltares/rtc-tools#1111]
In GitLab by @vreeken on Mar 26, 2019, 15:45

- ModelicaMixin implements `initial_state()` without calling `super.initial_state()`
- CSVMixin implements `initial_state()`, and calls `super().initial_state()`
- PIMixin implements `history()`, without calling `super().history()`.

Important:
OptimizationProblem has a `history()` implementation that defers to `initial_state()`, and an `initial_state()` implementation that defers to `history(). 

Using PIMixin will result in ModelicaMixin's `initial_state()` never being called. If we change PIMixin by doing `super().history()` there, we get stuck in an infinite recursion, because PIMixin also implements `initial_state()`. If we remove that one, it's fine. However, if we also let ModelicaMixin's `initial_state()` call `super().initial_state()`, it's back to infinite recursion.

Basically it would be prettier if all classes (other than OptimizationProblem) could just call their respective `super()` methods, whether they implement `initial_state` or `history`. Implementing both makes little sense in my opinion.

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on Mar 26, 2019, 15:50

mentioned in commit 103f171ebe1fbd23e12639113818897e6773f327

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on Apr 5, 2019, 16:52

mentioned in commit e3c6281508bd1e090f68d028b267a00a366233a6

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on Apr 5, 2019, 22:44

mentioned in commit 267a4c7b36d6fa7cc664df37995d1a103ac15b41

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on Apr 5, 2019, 23:09

mentioned in commit 391502da8f8c2fe3b3b37eacbb0e40a9aed61bc2

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on Apr 5, 2019, 23:37

closed via merge request !277

**Comment by SGeeversAtVortech** [comment]
In GitLab by @vreeken on Apr 5, 2019, 23:38

mentioned in commit 6e32d73fedd52a34447e782cd360c1d632e46d2c
